packages/engine/Source/Scene/Model/Extensions/Gpm/GltfMeshPrimitiveGpmLoader.js

```diff
@@ -0,0 +1,549 @@
+ *
+ * @private
+ * @experimental This feature is subject to change without Cesium's standard deprecation policy.
+ */
+function GltfMeshPrimitiveGpmLoader(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  const gltf = options.gltf;
+  const extension = options.extension;
+  const gltfResource = options.gltfResource;
+  const baseResource = options.baseResource;
+  const supportedImageFormats = options.supportedImageFormats;
+  const frameState = options.frameState;
+  const cacheKey = options.cacheKey;
+  const asynchronous = defaultValue(options.asynchronous, true);
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.gltf", gltf);
+  Check.typeOf.object("options.extension", extension);
+  Check.typeOf.object("options.gltfResource", gltfResource);
+  Check.typeOf.object("options.baseResource", baseResource);
+  Check.typeOf.object("options.supportedImageFormats", supportedImageFormats);
+  Check.typeOf.object("options.frameState", frameState);
+  //>>includeEnd('debug');
+
+  this._gltfResource = gltfResource;
+  this._baseResource = baseResource;
+  this._gltf = gltf;
+  this._extension = extension;
+  this._supportedImageFormats = supportedImageFormats;
+  this._frameState = frameState;
+  this._cacheKey = cacheKey;
+  this._asynchronous = asynchronous;
+  this._textureLoaders = [];
+  this._textureIds = [];
+  this._meshPrimitiveGpmLocal = undefined;
+  this._structuralMetadata = undefined;
+  this._state = ResourceLoaderState.UNLOADED;
+  this._promise = undefined;
+}
+
+if (defined(Object.create)) {
+  GltfMeshPrimitiveGpmLoader.prototype = Object.create(
+    ResourceLoader.prototype
+  );
+  GltfMeshPrimitiveGpmLoader.prototype.constructor = GltfMeshPrimitiveGpmLoader;
+}
+
+Object.defineProperties(GltfMeshPrimitiveGpmLoader.prototype, {
+  /**
+   * The cache key of the resource.
+   *
+   * @memberof GltfMeshPrimitiveGpmLoader.prototype
+   *
+   * @type {string}
+   * @readonly
+   * @private
+   */
+  cacheKey: {
+    get: function () {
+      return this._cacheKey;
+    },
+  },
+
+  /**
+   * The parsed GPM extension information from the mesh primitive
+   *
+   * @memberof GltfMeshPrimitiveGpmLoader.prototype
+   *
+   * @type {MeshPrimitiveGpmLocal}
+   * @readonly
+   * @private
+   */
+  meshPrimitiveGpmLocal: {
+    get: function () {
+      return this._meshPrimitiveGpmLocal;
+    },
+  },
+
+  /**
+   * Returns the result of converting the parsed 'MeshPrimitiveGpmLocal'
+   * into a 'StructuralMetadata'.
+   *
+   * Some details about the translation are intentionally not specified here.
+   *
+   * @memberof GltfMeshPrimitiveGpmLoader.prototype
+   *
+   * @type {StructuralMetadata}
+   * @readonly
+   * @private
+   */
+  structuralMetadata: {
+    get: function () {
+      return this._structuralMetadata;
+    },
+  },
+});
+
+GltfMeshPrimitiveGpmLoader.prototype.loadResources = async function () {
```

Comment:
For clarity, I might suggest marking functions used internally only with an underscore, ie. `GltfMeshPrimitiveGpmLoader.prototype._loadResources`.

Reply:
I now added some underscores there. On a certain level, this should not be necessary, because the whole class is tagged as `@private`, meaning that everything within this class already is private. And given that things like https://github.com/CesiumGS/cesium/blob/6c2e520420b95bcb6c8eba0f02c76347cee1dd4b/packages/engine/Source/Scene/ResourceLoader.js#L53 are marked as `private` and do _not_ have an undescore, I assume that "internally" basically means "in this file" here.

(This _can_ make sense, as a concept, but ... consistency is difficult. And I know that many of these difficulties can be attributed to the underlying "programming "language""...)